src/sugar3/activity/activity.py

```diff
@@ -1069,12 +1100,81 @@ def _keep_failed_dialog_response_cb(self, alert, response_id):
             self._session.will_quit(self, False)
 
     def can_close(self):
-        '''
-        Activities should override this function if they want to perform
-        extra checks before actually closing.
-        '''
+        """Activities should override this function if they want to perform
+        extra checks before actually closing."""
 
-        return True
```

Comment:
The reason 'ImageViewer' and 'Chat' didn't close is because these two activities override the 'can_close()' funtion of 'activitly.py' which is responsible to display my alert. 

I have figured out a solution and commit a new tweak for the same.